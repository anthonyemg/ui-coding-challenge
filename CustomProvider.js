import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducer from './src/public/reducers';

const store = createStore(reducer);

const CustomProvider = ({ children }) => {
  return <Provider store={store}>{children}</Provider>;
};

export default CustomProvider;
