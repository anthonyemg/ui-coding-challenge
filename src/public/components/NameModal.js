import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  name: PropTypes.string.isRequired,
  handleUpdateName: PropTypes.func.isRequired,
  handleSubmitUpdatedName: PropTypes.func.isRequired,
  handleToggleNameModal: PropTypes.func.isRequired,
  handleCancel: PropTypes.func.isRequired
};

const NameModal = ({
  name,
  handleUpdateName,
  handleSubmitUpdatedName,
  handleToggleNameModal,
  handleCancel
}) => (
  <div className="modal-content">
    <div>
      <span>Name</span>
      <input className="modal-input" defaultValue={name} onChange={handleUpdateName} />
    </div>

    <span className="modal-title">Edit Name</span>
    <div
      className="modal-closeButtonContainer"
      onClick={() => (handleCancel(), handleToggleNameModal())}
    >
      Close
      <button className="modal-closeButton">x</button>
    </div>
    <div className="modal-bottomButtons">
      <div className="modal-cancelButton" onClick={() => (handleCancel(), handleToggleNameModal())}>
        Cancel
      </div>
      <div
        className="modal-saveButton"
        onClick={() => (handleSubmitUpdatedName(), handleToggleNameModal())}
      >
        Save
      </div>
    </div>
  </div>
);

NameModal.propTypes = propTypes;

export default NameModal;
