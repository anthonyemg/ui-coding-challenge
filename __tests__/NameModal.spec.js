import React from 'react';
import renderer from 'react-test-renderer';
import CustomProvider from '../CustomProvider';
import NameModal from '../src/public/components/NameModal';

test('the NameModal component renders correctly', () => {
  const component = renderer.create(
    <CustomProvider>
      <NameModal />
    </CustomProvider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
