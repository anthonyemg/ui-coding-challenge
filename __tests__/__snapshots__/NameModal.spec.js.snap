// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the NameModal component renders correctly 1`] = `
<div
  className="modal-content"
>
  <div>
    <span>
      Name
    </span>
    <input
      className="modal-input"
      defaultValue={undefined}
      onChange={undefined}
    />
  </div>
  <span
    className="modal-title"
  >
    Edit Name
  </span>
  <div
    className="modal-closeButtonContainer"
    onClick={[Function]}
  >
    Close
    <button
      className="modal-closeButton"
    >
      x
    </button>
  </div>
  <div
    className="modal-bottomButtons"
  >
    <div
      className="modal-cancelButton"
      onClick={[Function]}
    >
      Cancel
    </div>
    <div
      className="modal-saveButton"
      onClick={[Function]}
    >
      Save
    </div>
  </div>
</div>
`;
